### PR TITLE
Shapeshift Integration

### LINKED ISSUE #96: Swap
### General
- [ ] Ensure that V4 implementation matches feature parity with V3
- [ ] Add Emoji in Navbar
- [ ] Show coin as disabled instead of hiding when unavailable

### Bity
- [x] ETH - > BTC
- [x] BTC -> ETH
- [x] ETH -> REP
- [x] BTC -> REP

### Shapeshift
- [ ] ETH Tokens -> ETH
- [ ] ETH -> ETH Tokens
- [ ] ETH -> Other coins

### Validations
- [ ] User is only allowed to send ETH if they have more ETH in the account they unlock than the swap they started
- [x] User is only allowed to send up to...
    - [x] BTC: Min: 0.01, Max: 3
    - [x] ETH: Derived from BTC
    - [x] REP: Derived from BTC
- [x] countdown stops at 0:00
- [x] Message: *Time has run out. \
    If you have already sent, please wait 1 hour. \
    If your order has not be processed after 1 hour, \
    please press the orange 'Issue with your Swap?' button.* appears when:
    - [x] Order cancelled
    - [x] Time has run out.
- [ ] Support Request Button
- [x] QR Code for BTC send
- [ ] address important todo in codebase about checking status in order status poll when order is not expired 

# Future

### Bity
- ETH / BTC -> EUR
- ETH / BTC -> CHF

### Coinbase
- USD -> ETH

### ShapeShift
- Other Coins -> ETH (lower priority)

### PR DESCRIPTION
**Pull request by eddiewang**
Closes #96 
### General
- [x] Ensure that V4 implementation matches feature parity with V3
- [x] Add Emoji in Navbar

### Bity
- [x] ETH - > BTC
- [x] BTC -> ETH
- [x] ETH -> REP
- [x] BTC -> REP

### Shapeshift
- [x] ETH Tokens -> ETH
- [x] ETH -> ETH Tokens
- [x] ETH -> Other coins

### Validations
- [x] (Requires Lite Send) User is only allowed to send ETH/Token if they have more ETH/Token in the account they unlock than the swap they started
- [x] User is only allowed to send up to...
    - [x] BTC: Min: 0.01, Max: 3
    - [x] ETH: Derived from BTC
    - [x] REP: Derived from BTC
- [x] countdown stops at 0:00
- [x] Message: *Time has run out. \
    If you have already sent, please wait 1 hour. \
    If your order has not be processed after 1 hour, \
    please press the orange 'Issue with your Swap?' button.* appears when:
    - [x] Order cancelled
    - [x] Time has run out.
- [x] Support Request Button
- [x] QR Code for BTC send
- [x] address important todo in codebase about checking status in order status poll when order is not expired 
- [x] Timer stops once order is complete.

### CODE REVIEW

File: common/api/bity.ts [hunk 1]
```diff
@@ -1,10 +1,34 @@
 import bityConfig, { WhitelistedCoins } from 'config/bity';
 import { checkHttpStatus, parseJSON, filter } from './utils';
+import bitcoinIcon from 'assets/images/bitcoin.png';
+import repIcon from 'assets/images/augur.png';
+import etherIcon from 'assets/images/ether.png';
 
 const isCryptoPair = (from: string, to: string, arr: WhitelistedCoins[]) => {
   return filter(from, arr) && filter(to, arr);
 };
```
Comment (eddiewang) [inline common/api/bity.ts]: Below are default fallback icon and name details for Bity swap pairs if Shapeshift fails.

File: common/api/bity.ts [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
 
+const btcOptions = {
```
Comment (eddiewang) [inline common/api/bity.ts]: Below are default fallback icon and name details for Bity swap pairs if Shapeshift fails.

File: common/sagas/swap/rates.ts [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
 import { TypeKeys } from 'actions/swap/constants';
 import { getAllRates } from 'api/bity';
 import { delay, SagaIterator } from 'redux-saga';
-import { call, cancel, fork, put, take, takeLatest } from 'redux-saga/effects';
+import { call, select, cancel, fork, put, take, takeLatest, race } from 'redux-saga/effects';
+import shapeshift from 'api/shapeshift';
+import { getSwap } from 'sagas/swap/orders';
 
 const POLLING_CYCLE = 30000;
+const SHAPESHIFT_TIMEOUT = 10000;
```
Comment (eddiewang) [inline common/sagas/swap/rates.ts]: Should we consider changing to a higher value?

Comment (dternyak) [reply common/sagas/swap/rates.ts]: If we want to get especially fancy, we could attempt to determine if the device is mobile / desktop, and increase the timeout on mobile.

For the MVP, I think 10s is fine.

File: common/sagas/swap/orders.ts [hunk 4]
```diff
@@ -97,12 +161,59 @@ export function* postBityOrderCreate(action: BityOrderCreateRequestedSwapAction)
...
+  yield put(loadShapeshiftRatesRequestedSwap());
+}
+
+export function* restartSwapSaga(): SagaIterator {
+  yield takeEvery(TypeKeys.SWAP_RESTART, restartSwap);
 }
 
-export function* bityTimeRemaining(): SagaIterator {
-  while (yield take('SWAP_ORDER_START_TIMER')) {
+export function* orderTimeRemaining(): SagaIterator {
```
Comment (dternyak) [inline common/sagas/swap/orders.ts]: This saga has turned into a monster. Can we break this out into separate generators?

File: common/containers/Tabs/Swap/components/LiteSend/LiteSend.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    this.props.configureLiteSend();
+  }
+
+  public render() {
+    if (!this.props.shouldDisplay) {
+      return null;
+    }
+    return (
+      <Aux>
+        <OfflineAwareUnlockHeader allowReadOnly={false} />
```
Comment (wbobeirne) [inline common/containers/Tabs/Swap/components/LiteSend/LiteSend.tsx]: I think we want to use WalletDecrypt instead. This adds a title that shouldn't be there, and you wouldn't be offline mid-swap anyway. The only thing we'd need to add is a button to change wallets into the Fields component.

**Pull request merged**